dotnet/src/Connectors/Connectors.OpenAI.UnitTests/Services/OpenAIChatCompletionServiceTests.cs

```diff
@@ -1036,6 +1036,76 @@ await foreach (var update in sut.GetStreamingChatMessageContentsAsync(chatHistor
         Assert.Equal("rainy", functionResult.GetProperty("content").GetString());
     }
 
+    [Fact]
+    public async Task ItCreatesCorrectFunctionToolCallsWhenUsingAutoFunctionChoiceBehaviorAsync()
```

Comment:
Would it be possible to create a base class for chat completion unit tests so that for each specific implementation we just need to write the code to instantiate the service and the prompt execution settings?

Not proposing we do this in the scope of this PR but if it's possible we could add a task to the backlog for this.

Reply:
Yes, that totally make sense. I'll add the base class either in a next PR or later as part of the "improvements" one.